src/asset-management/AaveManager.sol

```diff
@@ -129,9 +131,18 @@ contract AaveManager is IAssetManager, Owned(msg.sender), ReentrancyGuard {
         private
         returns (uint256 rShares)
     {
-        rShares = aAmount.divWad(_getExchangeRate(aAaveToken));
-        shares[aPair][aToken] -= rShares;
-        totalShares[aAaveToken] -= rShares;
+        rShares = aAmount.divWadUp(_getExchangeRate(aAaveToken));
+        if (shares[aPair][aToken] >= rShares) {
+            shares[aPair][aToken] -= rShares;
+        } else {
+            // clear it out
+            shares[aPair][aToken] = 0;
+        }
+        if (totalShares[aAaveToken] >= rShares) {
+            totalShares[aAaveToken] -= rShares;
+        } else {
+            totalShares[aAaveToken] = 0;
+        }
```

Comment:
Imagine scenario:

```txt
PAIR_A: 100 SHARES
PAIR_B: 100 SHARES

PAIR_A DOES FULL REDEEM
  - rShares = 101
  - shares[PAIR_A][TOKEN_A] -= 100 shares
  - totalSahres[PAIR_A][TOKEN_A] -= 101 shares
  
PAIR_B DOES FULL REDEEM
  - rShares = 101
  - shares[PAIR_B][TOKEN_A] -= 100 shares
  - totalSahres[PAIR_B][TOKEN_A] -= 99 shares
```

This does not result in locked funds but leads to the invariant `sum(balances) == totalSupply` breaking. I propose we instead do the following:
```solidity
        rShares = aAmount.divWadUp(_getExchangeRate(aAaveToken));
        if (rShares >= shares[aPair][aToken]) { rShares = shares[aPair][aToken]; }

        shares[aPair][aToken] -= rShares;
        totalShares[aAaveToken] -= rShares;
```

Reply:
good thinking. Implemented. The only thing I want to highlight is that the condition should be strictly greater, not greater or equal to 

`rShares > shares[aPair][aToken]`

Reply:
good catch, you're right